### PR TITLE
Refactors and adds tests for the data class

### PR DESCRIPTION
**Pull request by StephenNneji**
Added tests for the DataClass. Refactors both "DataClass.m" and calling routines in "projectClass.m"

### CODE REVIEW

**Review by DrPaulSharp — APPROVED** [review]

File: API/projectClass/dataClass.m [hunk 1]
```diff
@@ -297,65 +284,51 @@ function displayDataObject(obj)
...
-                    error('Data range and sim range must be [1 x 2] numeric arrays');
+                    throw(invalidType('Data range and sim range must be [1 x 2] numeric arrays'));
                 end
 
                 dataX = data(:,1);              % First column is always Q
                 realDataRange = [dataX(1),dataX(end)];
                 
                 if dataRange(1) < realDataRange(1)
-                    warning('Data range can''t be less than data min - resetting');
+                    warning('dataRange(1) can''t be less than min data value - resetting');
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply API/projectClass/dataClass.m:298]: I don't know of any MException equivalent for warnings

File: API/projectClass/dataClass.m [hunk 2]
```diff
@@ -297,65 +284,51 @@ function displayDataObject(obj)
...
                 if dataRange(2) > realDataRange(2)
-                    warning('Data range can''t be more than data max - resetting');
+                    warning('dataRange(2) can''t be more than max data value - resetting');
+                    dataRange(2) = realDataRange(2);
+                end
+                
+                if dataRange(1) > dataRange(2)
+                    warning('Data range min must not be greater than max - resetting');
+                    dataRange(1) = realDataRange(1);
+                    dataRange(2) = realDataRange(2);
```
[the remark this answers is not in the document]
Comment (StephenNneji) [reply API/projectClass/dataClass.m:310]: might be good to throw an exception if `realDataRange(1) > realDataRange(2)`

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply API/projectClass/dataClass.m:310]: Bingo! Sounds like a plan to me.

[the remark this answers is not in the document]
Comment (arwelHughes) [reply API/projectClass/dataClass.m:310]: Yup,  sounds good....

[the remark this answers is not in the document]
Comment (DrPaulSharp) [reply API/projectClass/dataClass.m:310]: invalidOption is for exceptions like "Model Type has to be Standard Layers, Custom Layers or Custom XY" - I'd keep that one as it is. This feels like a new exception to me, so maybe add invalidValue?

[the remark this answers is not in the document]
Comment (StephenNneji) [reply API/projectClass/dataClass.m:310]: An invalidOption is basically and invalidValue for an Enum but its okay to separate out

[the remark this answers is not in the document]
Comment (StephenNneji) [reply API/projectClass/dataClass.m:310]: Done

**Pull request merged**